summerparty/js/scripts.js

```diff
@@ -0,0 +1,103 @@
+class ImageScrollReveal extends HTMLElement {
+
+    #images = [];
+    #scrollSpeed = 1;
+    // Cache element height to improve performance
+    #imageHeight = 0;
+    #scrollPosition = 0;
+    #requestedAnimationFrame = null;
+
+    connectedCallback() {
+        this.#getScrollSpeed();
+        this.#getImages();
+        this.#measureImageHeight();
+        this.#setupResizeListener();
+        this.#setupScrollListener();
+        this.#setContainerHeight();
+    }
+
+    #getScrollSpeed() {
+        const speed = parseFloat(this.dataset.scrollSpeed);
+        this.#scrollSpeed = Number.isNaN(speed) ? 1 : speed;
+    }
+
+    #getImages() {
+        const selector = this.dataset.imagesSelector;
+        if (!selector) {
+            throw new Error('Required attribute data-images-selector is missing.');
+        }
+        this.#images = Array.from(this.querySelectorAll(selector));
+    }
+
+    #setupScrollListener() {
+        window.addEventListener('scroll', this.#handleScroll.bind(this));
+    }
+
+    #handleScroll(event) {
+        this.#scrollPosition = window.pageYOffset;
```

Comment:
Warum ist das nicht im AnimationFrame?

Reply:
Mh, weil wir lesen. Lesen immer ausserhalb des rAF, in die DOM schreiben immer innerhalb. Mit einem Read innerhalb des rAF invalidieren wir dieses. Bin mir allerdings nicht ganz sicher, ob wir mit pageYOffset effektiv lesen, but better safe than sorry.